packages/cli/init/src/utils/loadOpenApiFromUrl.ts

```diff
@@ -24,8 +24,23 @@ export interface FailedLoadOpenAPI {
 export async function loadOpenAPIFromUrl({ url, logger }: { url: string; logger: Logger }): Promise<LoadOpenAPIResult> {
     try {
         const response = await axios.get(url);
-        const jsonData = response.data;
-        const yamlData = dump(jsonData);
+        const contentType = response.headers["content-type"] ?? "";
+        if (typeof contentType !== "string") {
+            throw new Error("Content-Type from endpoint is not defined.");
+        }
+
+        let yamlData;
+
+        if (contentType.includes("json")) {
+            yamlData = dump(response.data);
+        } else if (contentType.includes("yaml")) {
+            yamlData = response.data;
+        } else {
+            throw new Error(
+                "Unsupported Content-Type from endpoint. Please ensure you're pointing to a URL that returns JSON or Y(A)ML and not HTML (e.g., Swagger UI webpage)"
+            );
+        }
```

Comment:
Nice! Mind putting this into a helper function so that it cleans up the control flow a bit? Plus, we might also want to be more permissive and start with a debug log instead of an error, just in case the Content-Type is empty for YAML content. In combination, I'm thinking of something like this:

```typescript
export async function loadOpenAPIFromUrl({ url, logger }: { url: string; logger: Logger }): Promise<LoadOpenAPIResult> {
    try {
        const yamlData = await fetchOpenAPIFromUrl({ url, logger });
        const tmpDir = await tmp.dir();
        const filePath = join(tmpDir.path, "openapi.yml");
        logger.debug("tmpDir", tmpDir.path);
        logger.debug("filePath", filePath);
        await writeFile(filePath, yamlData);
        return {
            status: LoadOpenAPIStatus.Success,
            filePath
        };
    } catch (error) {
        logger.debug(`Encountered an error while loading OpenAPI spec: ${JSON.stringify(error)}`);
        const errorMessage = `Failed to load OpenAPI spec from ${url}`;
        return {
            status: LoadOpenAPIStatus.Failure,
            errorMessage
        };
    }
}

async function fetchOpenAPIFromUrl({ url, logger }: { url: string; logger: Logger }): Promise<string> {
    const response = await axios.get(url);
    const contentType = response.headers["content-type"] ?? "";
    if (contentType.includes("json")) {
        return dump(response.data);
    }
    if (contentType.includes("yaml")) {
        return response.data;
    }
    logger.debug(`Unrecognized Content-Type "${contentType}" from endpoint. Please ensure you're pointing to a URL that returns JSON or YAML and not HTML (e.g. Swagger UI webpage)`);
    return response.data;
}
```

Reply:
Sounds good! I'm on it now 👍

Reply:
@amckinney I took your advice and implemented the suggestions! 

I also converted the `logger.debug` to a `logger.warn` so that users could see this warning message on their first run without having to try again with `--log-level debug`. Figured this could make for slightly better DX. If you'd prefer I swap this out for the original debug message, just let me know! ✌️